### PR TITLE
[MiHome] Add Aqara Lock & Static Code Analysis fixes

### PR DESCRIPTION
**Pull request by dimalo**
Signed-off-by: Dieter Schmidt <dima.loschkin@gmail.com>

- Add Aqara Fingerprint Lock
- fix some issues from static-code-analysis

### CODE REVIEW

File: addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_lock_aq1.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		<label>Aqara Fingerprint Door Lock</label>
+		<description>Live fingerprint unlock, password unlock, proximity card unlock, key unlock.</description>
+		<channels>
+			<channel id="status" typeId="status"></channel>
+			<channel id="id" typeId="idNumber"></channel>
+			<channel id="isOpen" typeId="alarm">
+				<label>Open Alarm</label>
+			</channel>
+			<channel id="wrongAccess" typeId="alarm">
+                <label>Wrong Access Alarm</label>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/ESH-INF/thing/sensor_lock_aq1.xml]: Please check the formatting of this xml file.

File: addons/binding/org.openhab.binding.mihome/README.md [hunk 2]
```diff
@@ -45,21 +41,20 @@ ___Coming soon___
     	1. Install MiHome app from [Google Play](https://play.google.com/store/apps/details?id=com.xiaomi.smarthome) or [AppStore](https://itunes.apple.com/app/mi-home-xiaomi-for-your-smarthome/id957323480) (your phone may need to be changed to English language first)
     	2.  In the app create a Mi Home account and make sure to set your region to Mainland (China) under Settings -> Locale
     	3.  If asked, update your gateway to the latest firmware (note that update window may pop up sequentially)
-
+
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/README.md]: Can you remove the spaces?

File: addons/binding/org.openhab.binding.mihome/README.md [hunk 3]
```diff
@@ -45,21 +41,20 @@ ___Coming soon___
...
-        7.  Make sure you hit the OK button (to the right of the cancel button) to save your changes
-
+    1.  Select your Gateway in the MiHome app
+    2.  Go to the "..." menu on the top right corner and click "About"
+    3.  Tap the version number "Plug-in version : 2.XX.X" at the bottom of the screen repeatedly until you enable developer mode
+    4.  You should now have 2 extra options listed: `wireless communication protocol` and `hub info`
+    5.  Choose `wireless communication protocol`
+    6.  Tap the toggle switch to enable WiFi functions. Note down the developer key (aka password), something like: 91bg8zfkf9vd6uw7
+    7.  Make sure you hit the OK button (to the right of the cancel button) to save your changes
+
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/README.md]: Can you remove the spaces?

File: addons/binding/org.openhab.binding.mihome/README.md [hunk 4]
```diff
@@ -412,29 +407,31 @@ sitemap xiaomi label="Xiaomi" {
...
 
 ## Supporting new devices
+
 The Xiaomi ecosystem grows at a steady rate. So there is a good chance that in the future even more devices get added to the suite. This section describes, how to get the necessary information to support new device types.
 
 ### Preconditions
+
 - you know how to access the [openHAB Console](https://www.openhab.org/docs/administration/console.html)
 - you have connected your gateway to openHAB and the communication is working
-
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/README.md]: Why did you remove this line?

File: addons/binding/org.openhab.binding.mihome/README.md [hunk 5]
```diff
@@ -412,29 +407,31 @@ sitemap xiaomi label="Xiaomi" {
...
 ### Enable debug logging for the binding
+
 - Enter ```log:set TRACE org.openhab.binding.mihome``` in the console to enable full logs.
 
     _When you are done you can disable the extended logging with ```log:set DEFAULT org.openhab.binding.mihome```_
 
 - Exit the console and start [viewing the logs](https://www.openhab.org/docs/tutorial/logs.html).
-
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/README.md]: Why did you remove this line?

**Comment by dimalo** [comment]
seems like it was too late yesterday... fixed all addressed issues.

File: addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml [hunk 6]
```diff
@@ -142,10 +142,10 @@
 		<state readOnly="true"></state>
 	</channel-type>
 
-	<channel-type id="isOpenAlarm">
+	<channel-type id="alarm">
 		<kind>trigger</kind>
-		<label>Opened window alarm</label>
-		<description>Triggers ALARM event, when window sensor is not getting "closed" for configured amount of time</description>
+		<label>Alarm trigger</label>
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml]: We have the (undocumented...) best practice to keep labels always capitalised: "Alarm Trigger".

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorLockHandler.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    @Override
+    void parseReport(JsonObject data) {
+        super.parseDefault(data);
+    }
+
+    @Override
+    void parseReadAck(JsonObject data) {
+        if (data.has(FING_VERIFIED)) {
+            onOpen();
+            updateState(CHANNEL_STATUS, new StringType("Finger"));
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorLockHandler.java]: Does it make sense to use a String constant for "Finger" too?

File: addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorLockHandler.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+            updateState(CHANNEL_ID, new DecimalType(data.get(CARD_VERIFIED).getAsInt()));
+        } else if (data.has(VERIFIED_WRONG)) {
+            updateState(CHANNEL_STATUS, new StringType("Wrong Access"));
+            updateState(CHANNEL_ID, new DecimalType(data.get(VERIFIED_WRONG).getAsInt()));
+            triggerChannel(CHANNEL_WRONG_ACCESS, "ALARM");
+        }
+    }
+
+    private void onOpen() {
+        triggerChannel(CHANNEL_IS_OPEN, "ALARM");
```
Comment (cweitkamp) [inline addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/XiaomiSensorLockHandler.java]: See above.

**Comment by cweitkamp** [comment]
@dimalo Did you see my comments? Would be nice if we can make some progress an finalize this PR.

**Comment by dimalo** [comment]
@cweitkamp sorry you had to wait so long for just a few changes. Back from a businesstrip and using the jetlag for it 😄

**Comment by RUSRaptor** [comment]
Hi! Is it possible to make Aqara Lock S2 support?
